libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -542,10 +542,15 @@ bool VoteManager::checkRewardVotes(const std::shared_ptr<PbftBlock>& pbft_block)
   return true;
 }
 
-void VoteManager::replaceRewardVotes(std::vector<std::shared_ptr<Vote>>&& cert_votes) {
-  if (cert_votes.empty()) return;
+std::vector<std::shared_ptr<Vote>> VoteManager::replaceRewardVotes(
+    const std::vector<std::shared_ptr<Vote>>& cert_votes) {
+  if (cert_votes.empty()) return {};
 
   std::unique_lock lock(reward_votes_mutex_);
+  std::vector<std::shared_ptr<Vote>> reward_votes;
+  for (const auto& v : reward_votes_) {
+    reward_votes.push_back(v.second);
```

Comment:
here we should do move as we are replacing it later, but not like this is should be `reward_votes = std::move(reward_votes_);`